lib/types.js

```diff
@@ -55,6 +55,19 @@ var LOGICAL_TYPE = null;
 // to be able to reference names (i.e. for branches) during instantiation.
 var UNDERLYING_TYPES = [];
 
+
+// range of values represented by 32 bit signed int
+const MAX_INT = 2 ** 31 - 1;
+const MIN_INT = 2 ** 31 * -1;
+
+// range of values represented by 64 bit signed int
+const MAX_LONG = 2 ** 63 - 1;
+const MIN_LONG = 2 ** 63 * -1;
+
+// range of values possible to represent by IEEE 754 single precision floating point fraction part
+const MAX_SAFE_FLOAT = 2 ** 23 - 1;
+const MIN_SAFE_FLOAT = -MAX_SAFE_FLOAT;
```

Comment:
This is misleading - floats can also underflow faster than doubles. This is not a safe range. I'm not sure there's a good way to efficiently and correctly infer float vs double though. I recommend we either keep the existing code as-is or (better but optional) simplify to always inferring double on non-integer and unsafe-long input.